cron.go

```diff
@@ -540,114 +547,135 @@ func (c *Cron) run(ctx context.Context) {
 
 		case now = <-time.After(effective.Sub(now)):
 			// Run every entry whose next time was this effective time.
+			currentWindow := time.Now().Truncate(time.Second)
+
 			for _, e := range entries {
-				if e.Next != effective {
+				if e.next != effective {
 					break
 				}
-				e.Prev = e.Next
+				e.prev = e.next
+
 				e.tick(effective)
 
-				go func(ctx context.Context, e *Entry, next time.Time) {
+				// Copy since we don't need to reference
+				job := e.Job
+
+				req := TriggerRequest{
+					JobName:   job.Name,
+					Timestamp: effective,
+					Actual:    time.Now(),
+					Metadata:  job.Metadata,
+					Payload:   job.Payload,
+				}
+
+				deleteJobCallback := func(ctx context.Context) error {
+					return deleteJob(ctx, job)
+				}
+
+				if job.Status == JobStatusDeleted || e.Job.expired(effective) {
+					_, deleted := c.deletedJobs.Load(job.Name)
+					if !deleted {
+						// No cron hard-deleted it yet from the db, still just a soft delete.
+						c.deletedJobs.Store(job.Name, true)
+						c.jobCollector.Add(deleteJobCallback)
+						// Won't trigger.
+						continue
+					}
+				}
+
+				// If it is an old trigger (not in the same second window), we skip.
+				// Delays of the trigger function should not delay this step.
+				if currentWindow != effective {
```

Comment:
This is what fixed the "duplicate" triggers that @cicoyle was seeing - those were overdue triggers because the cron scheduler is "stuck" in the past at start if there is no job triggering.